docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+            data_window = norm_train_data_cols[row:row + timestep]
+            data_vector = data_window.flatten()
+            state = add_labels(row + timestep, row + timestep + output_timestep, df_train)
+            train_pairs.append((data_vector, state))
+            saved_step = row + timestep
+        else:
+            break
+
+Now that we have the data pairs we can start creating our x/y training data frames.
+
+.. code-block:: python
+
+    import numpy as np
+
+
+    input_len = train_pairs[0][0].shape[0]
+
+    x_train = np.zeros((len(train_pairs), input_len))
+    y_train = np.zeros((len(train_pairs),))
+
+    for index in range(0, len(train_pairs)):
+        (x, y) = train_pairs[index]
+        x_train[index, 0:input_len] = x
+        y_train[index] = y
+
+After creating our training data frames we can now continue with creating the MLP model.
+We are going to use 3 layers, with the `ReLU`_ activation function for the first two
+layers. If the model should get worse while going through the epochs we will stop the
+training with the ``EarlyStopping`` callback.
+
+.. code-block:: python
+
+    import tensorflow as f
+    import matplotlib.pyplot as plt
+    from tensorflow.python.keras.callbacks import EarlyStopping
+    from tensorflow.keras import layers
+
+
+    mlp_model = tf.keras.Sequential()
+
+    mlp_model.add(layers.Dense(200, activation="relu", input_shape=(input_len,)))
+    mlp_model.add(layers.Dense(40, activation="relu"))
+    mlp_model.add(layers.Dense(1))
+
+    mlp_model.compile(optimizer="adam",
+                      loss=tf.keras.losses.MeanSquaredError(),
+                      metrics=["accuracy"])
+
+    es = EarlyStopping(monitor="loss", mode="auto", verbose=1, patience=50)
+    hist = mlp_model.fit(x_train, y_train, epochs=500, callbacks=[es])
+
+    plt.plot(hist.history["loss"])
+    plt.show()
+
+.. figure:: mlp_model_train_loss.png
+   :align: left
+
+Now that we have our MLP model, we can start preparing the test data to verify
+the accuracy of the model.
+
+.. code-block:: python
+
+    test_pairs = []
+
+    test_saved_step = 0
+    for i in df_test.index:
+        row = test_saved_step
+        if row + timestep + output_timestep < len(df_test.index):
+            data_window = norm_test_data_cols[row:row + timestep]
+            data_vector = data_window.flatten()
+            state = add_labels(row + timestep, row + timestep + output_timestep, df_test)
+            test_pairs.append((data_vector, state))
+            test_saved_step = row + timestep
+        else:
+            break
+
+    x_test = np.zeros((len(test_pairs), input_len))
+    y_test = np.zeros((len(test_pairs),))
+
+    for index in range(0, len(test_pairs)):
+        (x, y) = test_pairs[index]
+        x_test[index, 0:input_len] = x
+        y_test[index] = y
+
+The data is prepared in the same way as the training data was prepared, and now we can
+predict the test data and look at the accuracy.
```

Comment:
```suggestion
This data is prepared in the same way as the training data. It can now be used with the model
to generate predictions and to estimate the model's accuracy.
```